### PR TITLE
Guidelines for writing bash

### PR DESCRIPTION
**Pull request by raoulmillais**
What I am suggesting:

- [x] Gets out of the way;
- [x] Is not restricting;
- [x] Is safe, production ready, and project members have experience with it;
- [x] Promotes good coding practices.

- Hat tip to @sublimino for some of this advice

### CODE REVIEW

File: bash-code-style.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Writing Bash Scripts
+
+## When to use bash
+
+If you are coordinating a bunch of other utilities and it is not a complex
+process bash is a good choice.  It can also be helpful for prototyping quickly
+before reaching for a more powerful language.  As a rule of thumb bash scripts
+should never grow larger than a few hundred lines.  You probably want a better
+language at that point.
```
Comment (tomgco) [inline bash-code-style.md]: Maybe a bit too opinionated, maybe  "You probably want a better language at that point." could be rephrased to: "At this stage you might want to re-evaluate choosing bash"

File: bash-code-style.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+## Bash code style
+
+In general follow [google's excellent recommendations](https://google.github.io/styleguide/shell.xml)
```
Comment (tomgco) [inline bash-code-style.md]: google's -> Google's

**Review by fabiosantoscode — APPROVED** [review]
I've nothing to add here, though I second Tom's comments :O

**Comment by sublimino** [comment]
- Include testing framework https://github.com/sstephenson/bats and https://github.com/ztombol/bats-assert
- set script to own directory can be useful for relieve path includes `cd "$(dirname "$0")"`
- I've got an options parser I'll link in
- A standard library would be useful, mine's currently in flux so this is perhaps a spike
- I'm using a yeoman generator for scripts atm too - this auto generates a usage from comments in the file. Can paste in when not on mobile if interested
- couldn't see debug mode mentioned at first skim, can be handy `set -x`
- there's a BASH debugger if things get hairy http://bashdb.sourceforge.net
- osx still ships with v3, could be worth highlighting
- I <3 BASH

File: bash-code-style.md (hunk 1 above)
Comment (raoulmillais) [reply bash-code-style.md]: Yep I see your point. It's verges towards flamey for a standards document. :+1:

File: bash-code-style.md (hunk 2 above)
Comment (raoulmillais) [reply bash-code-style.md]: :+1: Thanks

**Comment by raoulmillais** [comment]
Thanks @tomgco and @sublimino.  I incorporated your suggestions.  @sublimino I incorporated your debugging tips, please do submit a pull request to expand and add any other tips :)

**Review by tomgco — APPROVED** [review]
😲 Awesome.

**Pull request merged**